src/webServiceClient.ts

```diff
@@ -128,78 +126,84 @@ export default class WebServiceClient {
       } as WebServiceClientError);
     }
 
-    const options = {
-      auth: `${this.accountID}:${this.licenseKey}`,
+    const controller = new AbortController();
+    const timeoutId = setTimeout(() => controller.abort(), this.timeout);
+    const options: RequestInit = {
       headers: {
         Accept: 'application/json',
+        Authorization:
+          'Basic ' +
+          Buffer.from(`${this.accountID}:${this.licenseKey}`).toString(
+            'base64'
+          ),
         'User-Agent': `GeoIP2-node/${version}`,
       },
-      host: this.host,
       method: 'GET',
-      path: parsedPath,
-      timeout: this.timeout,
+      signal: controller.signal,
     };
 
     return new Promise((resolve, reject) => {
-      const req = https.request(options, (response) => {
-        let data = '';
-
-        response.on('data', (chunk) => {
-          data += chunk;
-        });
-
-        response.on('end', () => {
+      return fetch(url, options)
+        .then(async (response) => {
+          let data;
           try {
-            data = JSON.parse(data);
+            data = await response.json();
           } catch {
             return reject(this.handleError({}, response, url));
           }
 
-          if (response.statusCode && response.statusCode !== 200) {
+          if (response.status && response.status !== 200) {
             return reject(
               this.handleError(data as ResponseError, response, url)
             );
           }
 
           return resolve(new modelClass(data));
+        })
+        .catch((err: NodeJS.ErrnoException) => {
+          if (err.name === 'AbortError') {
+            return reject({
+              code: 'FETCH_TIMEOUT',
```

Comment:
I don't want to introduce a new error code here, but `ERR_HTTP_REQUEST_TIMEOUT` doesn't seem correct: "The client has not sent the entire request within the allowed time."

Using `ETIMEDOUT` might be confusing as this error is not being raised by node itself.

Open to suggestions.

Reply:
`ABORT_ERR` seems appropriate here. https://nodejs.org/api/errors.html#abort_err

Reply:
Actually, we pretty much have free reign here for error codes and messages